### PR TITLE
feature(Reachability): add connection checking for datafile-fetch and event-dispatch

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
- Add connection constraints to WorkManager requests for datafile-fetch and event-dispatch
- Fix default event-dispatch retry to disabled (configurable)

## Test plan
- Unit tests for DatafileWorker and EventWorker 
- Manual tests with devices

## Issues
-  OASIS-8025

### CODE REVIEW

File: android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyManager.java [hunk 1]
```diff
@@ -898,10 +897,10 @@ public OptimizelyManager build(Context context) {
 
             if (datafileDownloadInterval > 0) {
                 // JobScheduler API doesn't allow intervals less than 15 minutes
-//                if (datafileDownloadInterval < 900) {
-//                    datafileDownloadInterval = 900;
-//                    logger.warn("Minimum datafile polling interval is 15 minutes. Defaulting to 15 minutes.");
-//                }
+                if (datafileDownloadInterval < 900) {
```
Comment (The-inside-man) [inline android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyManager.java:900]: Should we define a constant for 900?

**Review by The-inside-man — APPROVED** [review]
LGTM! Just one small comment.

File: shared/src/androidTest/java/com/optimizely/ab/android/shared/WorkerSchedulerTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
+package com.optimizely.ab.android.shared;
```
Comment (mnoman09) [inline shared/src/androidTest/java/com/optimizely/ab/android/shared/WorkerSchedulerTest.java:1]: header is missing.

**Review by mnoman09 — APPROVED** [review]
Overall looks good just some headers are missing and some needs to be updated. Otherwise everything LGTM.

**Pull request merged**